streams/hash/sha3/sha3_interface.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <cstdint>
+#include <streams/hash/hash_interface.h>
+
+namespace sha3 {
+
+using BitSequence = unsigned char;
```

Comment:
I think it would be good to avoid repeatedly declaring BitSequence and DataLength, it should be only in one place so that it would be easy to change its value.